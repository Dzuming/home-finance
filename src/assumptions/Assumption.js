import React, { Component } from 'react';
import Nav from './Nav';
import MonthlyAssumption from './MonthlyAssumption';
import OverallAssumptions from './OverallAssumptions';

const TABS = {
  0: () => <MonthlyAssumption />,
  1: () => <OverallAssumptions />,
};

class Assumption extends Component {
  state = {
    currentTab: 0,
  };

  handleTabChange = (event, value) => this.setState({ currentTab: value });

  render() {
    const { currentTab } = this.state;
    return (
      <React.Fragment>
        <Nav currentTab={currentTab} handleTabChange={this.handleTabChange} />
        {TABS[currentTab]()}
      </React.Fragment>
    );
  }
}

export default Assumption;
