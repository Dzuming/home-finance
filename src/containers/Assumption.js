import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Assumption extends Component {
  render() {
    return <div>test</div>;
  }
}

Assumption.propTypes = {};

export default Assumption;
