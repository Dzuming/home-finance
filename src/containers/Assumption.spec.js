import React from 'react';
import { create } from 'react-test-renderer';
import Assumption from './Assumption';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const store = mockStore({
  budget: 0,
  user: [],
  login: { isAuthenticated: true },
});
const component = create(
  <Provider store={store}>
    <Assumption />
  </Provider>,
);

describe('<Assumption />', () => {
  it('should match snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
