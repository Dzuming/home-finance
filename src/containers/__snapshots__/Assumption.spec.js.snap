// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Assumption /> should match snapshot 1`] = `
<div
  className="MuiGrid-typeContainer-1"
>
  <div
    className="MuiGrid-typeItem-2 MuiGrid-grid-xs-4-30"
  >
    <div
      className="MuiPaper-root-91 MuiPaper-shadow2-95 MuiPaper-rounded-92"
    >
      <div
        className="MuiCardContent-root-118"
      >
        <aside
          className="MuiTypography-root-119 MuiTypography-body2-127 MuiTypography-gutterBottom-136 MuiTypography-alignCenter-132"
        >
          Assumption
        </aside>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-typeItem-2 MuiGrid-grid-xs-4-30"
  >
    <div
      className="MuiPaper-root-91 MuiPaper-shadow2-95 MuiPaper-rounded-92"
    >
      <div
        className="MuiCardContent-root-118"
      >
        <aside
          className="MuiTypography-root-119 MuiTypography-body2-127 MuiTypography-gutterBottom-136 MuiTypography-alignCenter-132"
        >
          percentage
        </aside>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-typeItem-2 MuiGrid-grid-xs-4-30"
  >
    <div
      className="MuiPaper-root-91 MuiPaper-shadow2-95 MuiPaper-rounded-92"
    >
      <div
        className="MuiCardContent-root-118"
      >
        <aside
          className="MuiTypography-root-119 MuiTypography-body2-127 MuiTypography-gutterBottom-136 MuiTypography-alignCenter-132"
        >
          Value
        </aside>
      </div>
    </div>
  </div>
</div>
`;
